### ISSUE TITLE
Panic in `FromStr<DateTime<Utc>>`

### ISSUE THREAD

**Issue by jjbayer** [chronotope/chrono#1253]
Special characters in date strings produce a panic (reproduced in 0.4.27 and 0.4.28):

```rust
#[test]
fn chrono_panics() {
    "2015-02-18T23:16:9.15øøø".parse::<DateTime<Utc>>();
}
 ```

The bug was possibly introduced in https://github.com/chronotope/chrono/pull/1145, which uses [unchecked string slicing](https://github.com/pitdicker/chrono/blob/012c3c8dd9419f552fdc51b5bb031ba0b42027eb/src/format/parse.rs#L589).

**Comment by pitdicker** [comment]
Thank you for the report, I should know better than to do arbitrary string slicing.
Will be fixed with https://github.com/chronotope/chrono/pull/1254.

**Comment by jjbayer** [comment]
@pitdicker Awesome, thank you! Any chance this will make it into a release soon?

**Comment by pitdicker** [comment]
Published as 0.4.29.
